File: frontend/src/api/students.ts
import { api } from "./client";

export interface StudentListItem {
  enroll_no: string;
  name: string | null;
  father_name: string | null;
  category: string | null;
  gender: string | null;
  mobile: string | null;
  email: string | null;
}

export interface EnrollmentOut {
  session: string | null;
  course_id: string | null;
  course_name: string | null;
  semester: string | null;
  roll_no: string | null;
  student_type: string | null;
  status: string | null;
  form_status: string | null;
}

export interface StudentProfile {
  enroll_no: string;
  name: string | null;
  name_hindi: string | null;
  father_name: string | null;
  mother_name: string | null;
  gender: string | null;
  dob: string | null;
  category: string | null;
  ews: string | null;
  mobile: string | null;
  email: string | null;
  address1: string | null;
  city1: string | null;
  state1: string | null;
  photo_url: string | null;
  sign_url: string | null;
  prof_status: string | null;
  enrollments: EnrollmentOut[];
}

export async function listStudents(search: string, limit = 50, offset = 0): Promise<StudentListItem[]> {
  const { data } = await api.get<StudentListItem[]>("/students", {
    params: { search: search || undefined, limit, offset },
  });
  return data;
}

export async function getStudent(enroll: string): Promise<StudentProfile> {
  const { data } = await api.get<StudentProfile>(`/students/${encodeURIComponent(enroll)}`);
  return data;
}

File: backend/app/schemas/students.py
"""Schemas for students + academic master data."""
from datetime import date

from pydantic import BaseModel


class StudentListItem(BaseModel):
    enroll_no: str
    name: str | None = None
    father_name: str | None = None
    category: str | None = None
    gender: str | None = None
    mobile: str | None = None
    email: str | None = None


class EnrollmentOut(BaseModel):
    session: str | None = None
    course_id: str | None = None
    course_name: str | None = None
    semester: str | None = None
    roll_no: str | None = None
    student_type: str | None = None
    status: str | None = None
    form_status: str | None = None


class StudentProfile(BaseModel):
    enroll_no: str
    name: str | None = None
    name_hindi: str | None = None
    father_name: str | None = None
    mother_name: str | None = None
    gender: str | None = None
    dob: date | None = None
    category: str | None = None
    ews: str | None = None
    mobile: str | None = None
    email: str | None = None
    address1: str | None = None
    city1: str | None = None
    state1: str | None = None
    photo_url: str | None = None
    sign_url: str | None = None
    prof_status: str | None = None
    enrollments: list[EnrollmentOut] = []


class SessionOut(BaseModel):
    session: str
    status: str | None = None


class CourseOut(BaseModel):
    course_id: str
    course_name: str | None = None
    course_type: str | None = None
    faculty: str | None = None
    scheme: str | None = None
    status: str | None = None


class CollegeOut(BaseModel):
    college_id: str
    college_name: str | None = None
    college_type: str | None = None
    city: str | None = None
    status: str | None = None


class AcademicDashboard(BaseModel):
    session: str | None = None
    total_enrollments: int
    active_enrollments: int
    exam_forms_paid: int
